CPS-????/README.md

```diff
@@ -0,0 +1,173 @@
+---
+CPS: ?
+Title: Intents for Cardano
+Status: Open
+Category: ?
+Authors:
+    - Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Proposed Solutions: []
+Discussions:
+Created: 2024-03-11
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We describe a family of issues around “intents” on Cardano. We do some basic analysis on the problem space, and suggest some high-level areas for investigation.
+
+## Problem
+
+In many cases, a user’s desired interaction with a system is not fully specified. 
+Classic examples include things like “market orders” where a user wants to buy an asset at the “market price”, and does not specify exactly what that is, leaving it up to another system to resolve. 
+A common name for these kinds of “incomplete” or “partial” actions is “intents”. 
+We refer to the process of turning an intent into something complete as “fulfilment”.
+
+Intents are [quite prevalent](https://members.delphidigital.io/reports/wtf-is-anoma-part-2-wtf-are-intent-based-apps) in blockchain systems (see “Use cases” for more discussion of specific examples), but they are also problematic since blockchain systems usually require a fairly high degree of specificity in their settlement layer. 
+For example, if you want to perform a trade, you usually need to specify exactly who with and exactly how much for.
```

Comment:
i feel like this isnt really the case for account-based systems, at least

Reply:
You still need to say who you're sending money to and how much money you're spending! 

I do think there's an argument that smart contracts on Ethereum effectively support a limited kind of intent-based interaction where you tell the smart contract the action you want to take and then it implements it, potentially non-deterministically.

Reply:
yeah, this is what i mean - systems where you don't know what state your tx will be applied to are already kind of intent-processing systems. The only argument for less on-chain intent processing here is that it will be cheaper/faster.

Reply:
Yeah, and also that your intents don't get "matched", there is a single provider which is the smart contract. I think it's maybe less confusing to not call those "intents" and just say it's non-deterministic transaction processing, but IDK.

Reply:
Saying that, in general, we want intent processing on systems because we want that functionality **added**, rather than that "we want to do it cheaper and faster and easier" is a bit misleading, so i would reword.